psi4/src/psi4/libmints/sieve.h

```diff
@@ -1,276 +0,0 @@
- *
- * The copyrights for code used from other parties are included in
- * the corresponding files.
- *
- * This file is part of Psi4.
- *
- * Psi4 is free software; you can redistribute it and/or modify
- * it under the terms of the GNU Lesser General Public License as published by
- * the Free Software Foundation, version 3.
- *
- * Psi4 is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public License along
- * with Psi4; if not, write to the Free Software Foundation, Inc.,
- * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
- *
- * @END LICENSE
- */
-
-#ifndef SIEVE_H
-#define SIEVE_H
-
-// need this for erfc^{-1} in the QQR sieve
-//#include <cfloat>
-#include <vector>
-#include <memory>
-//#include <utility>
-#include "psi4/pragma.h"
-#include "psi4/libmints/vector3.h"
-
-namespace psi {
-
-class BasisSet;
-
-/**
- * ERISieve
- *
- * Class to perform ERI sieving
- *
- * At the moment, this class uses Cauchy-Schwarz sieving,
- * and is designed to be used in concert with density-based
- * sieving. In the future, it is hoped that MBIE sieving may
- * be added to this class.
- *
- * Use of this class is as follows:
- *
- *     // Initialize the sieve object
- *     std::shared_ptr<ERISieve> sieve(basisset, sieve_cutoff);
- *
- *     // Reset the sieve cutoff (you can do this wherever)
- *     sieve->set_sieve(new_cutoff);
- *
- *     ...
- *
- *     // Investigate stuff. This example is for shells, all methods are
- *     // also provided for functions
- *
- *     // Compute a shell quartet (MN|RS), if (MN|RS) >= sieve_cutoff
- *     if (sieve->shell_significant(M,N,R,S)) eri->compute(M,N,R,S);
- *
- *     // Compute a shell quartet (MN|RS), if (MN|RS) * D_RS >= sieve_cutoff
- *     // Squares are used to avoid sqrt()
- *     if (sieve->shell_ceiling2(M,N,R,S) * D_RS * D_RS >= sieve_cutoff * sieve_cutoff)
- *         eri->compute(M,N,R,S);
- *
- *     // Index the significant MN shell pairs (triangular M,N)
- *     const std::vector<std::pair<int,int> >& MN = sieve->shell_pairs();
- *     for (long int index = 0L; index < MN.size(); ++index) {
- *         int M = MN[index].first;
- *         int N = MN[index].second;
- *     }
- *
- *     // Check if a triangular index MNindex (M * (M + 1) / 2) + N exists,
- *     // and if so, where it starts in reduced triangular MN
- *     int MNindex = (M * (M + 1) >> 1) + N;
- *     const std::vector<long int> >& MN_reverse = sieve->shell_pairs_reverse();
- *     int MNreduced = MN_reverse[MNindex];
- *     if (MNreduced < 0) {
- *         // The shell pair is not signficant
- *     } else {
- *         // The shell pair is the MNreduced significant shell pair
- *     }
- *
- *
- */
-
-class PSI_API __attribute__((deprecated(
```

Comment:
Since this PR removes a class marked as `PSI_API`, I think it should be mentioned in the API changes section of the PR.

Reply:
Good point. Done and done!